### DISCUSSION TITLE
How can i get the utf-8 filename（The filename is garbled）

### DISCUSSION THREAD

**Discussion by liyunsong** [101arrowz/fflate#153] [Q&A]
How can i get the utf-8 filename

Unzip chinese language filename 

```javascript
const unZipper = new Unzip()
  unZipper.onfile = async (f) => {
  console.log(f.name)
}
```

<img width="306" alt="image" src="https://user-images.githubusercontent.com/6094746/205131666-c55172e6-5524-4688-a44e-34f60b135d29.png">

the right is "111liyunsong/3松.webp"

**Comment by 101arrowz** [comment]
`fflate` seems to be using ISO-8859-1 decoding instead of UTF-8, though it supports UTF-8 when specified in the file header. Could you send me a reproduction ZIP so I can take a look?

**Reply by liyunsong** [reply]
thanks  a lot
[111liyunsong.zip](https://github.com/101arrowz/fflate/files/10138502/111liyunsong.zip)

**Reply by 101arrowz** [reply]
Unfortunately it looks like macOS [has issues with ZIP filename encoding](https://stackoverflow.com/questions/13261347/correctly-decoding-zip-entry-file-names-cp437-utf-8-or). I don't think this should be fixed in `fflate` as it's parsing the ZIP files completely according to the specification. However, I do have a way for you to recover the original filenames:

```js
import { Unzip, strFromU8, strToU8 } from 'fflate';

const unzipper = new Unzip();
unzipper.onfile = async (f) => {
    // Only do this if you're sure it's a macOS ZIP
    let name = strFromU8(strToU8(f.name, true));
}
```

**Reply by liyunsong** [reply]
I do not know why 'STR-U8 turn and turn'  works good. but it is not universal solution。

this case works not good

[fflate.zip](https://github.com/101arrowz/fflate/files/10141181/fflate.zip)  fflate/新建 文本文档.txt
